src/osip-opcua-client/src/main/java/edu/kit/pse/osip/core/opcua/client/UAClientWrapper.java

```diff
@@ -168,20 +166,11 @@ public void connectClient() throws UAClientException {
         doSubscribe(Identifiers.Server_ServerStatus_CurrentTime, CONNECTION_TIMEOUT / 2,
                 (IntReceivedListener) value -> { }, Identifiers.DateTime);
 
-        client.getSubscriptionManager().addSubscriptionListener(new SubscriptionListener() {
-            @Override
-            public void onPublishFailure(UaException exception) {
-                connected = false;
-                if (errorListener != null) {
-                    errorListener.onError(ERROR_DISCONNECT);
-                }
-            }
-        });
         client.addSessionActivityListener(new SessionActivityListener() {
             @Override
             public void onSessionInactive(UaSession session) {
-                connected = false;
-                if (errorListener != null) {
+                if (connected && errorListener != null) {
+                    connected = false;
```

Comment:
Warum wurde das `connected = false` in die Abfrage geschoben? Was wenn der errorListener null ist? Dann wird connected nicht auf false gesetzt.